lib/dialects/sqlite3/schema/ddl.js

```diff
@@ -359,6 +359,106 @@ assign(SQLite3_DDL.prototype, {
     );
   },
 
+  primary: async function (columns, constraintName) {
+    return this.client.transaction(
+      async (trx) => {
+        this.trx = trx;
+
+        const tableInfo = (await this.getTableSql())[0];
+        const currentSQL = tableInfo.sql;
+
+        const oneLineSQL = currentSQL.replace(/\s+/g, ' ');
+        const matched = oneLineSQL.match(/^CREATE TABLE\s+(\S+)\s*\((.*)\)/);
+
+        const columnDefinitions = matched[2];
+
+        if (!columnDefinitions) {
+          throw new Error('No column definitions in this statement!');
+        }
+
+        let newColumnDefinitions = columnDefinitions
+          .split(/,(?![^(]*\))/g) // comma, except when in parentheses
+          .map((line) => line.trim())
+          .filter((line) => line.startsWith('primary') === false)
+          .map((line) => line.replace(/primary key/i, ''));
+
+        let primaryKeyDef = `primary key(${columns.join(',')})`;
+
+        if (constraintName) {
+          primaryKeyDef = `constraint ${constraintName} ${primaryKeyDef}`;
+        }
+
+        newColumnDefinitions.push(primaryKeyDef);
+        newColumnDefinitions = newColumnDefinitions.join(', ');
+
+        const newSQL = oneLineSQL.replace(
+          columnDefinitions,
+          newColumnDefinitions
+        );
+
+        return this.reinsertMapped(tableInfo, newSQL, (row) => {
+          return row;
+        });
+      },
+      { connection: this.connection }
+    );
+  },
+
+  foreign: async function (foreignInfo) {
+    return this.client.transaction(
+      async (trx) => {
+        this.trx = trx;
+
+        const tableInfo = (await this.getTableSql())[0];
+        const currentSQL = tableInfo.sql;
+
+        const oneLineSQL = currentSQL.replace(/\s+/g, ' ');
```

Comment:
I see lots of overlap with previous method. maybe some code can be moved to a shared internal function?